drivers/can/can_mcp2515.c

```diff
@@ -358,6 +385,16 @@ static void mcp2515_detach(struct device *dev, int filter_nr)
 	k_mutex_unlock(&dev_data->filter_mutex);
 }
 
+static void mcp2515_register_state_change_isr(struct device *dev,
+						can_state_change_isr_t isr)
+{
+	struct mcp2515_data *dev_data = DEV_DATA(dev);
+
+	k_mutex_lock(&dev_data->tx_mutex, K_FOREVER);
```

Comment:
`tx_mutex` previously was just a mutex for the `tx_busy_map` for send calls. The rx functions (`attach_isr` and `detach` use `filter_mutex`). I think using `tx_mutex` for more is fine, but wanted to point out it isn't generally used as a mutex to accessing the peripheral.

Reply:
There is a global mutex now.